build-entities.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+set -e
+
+# Absolute path to the directory containing this script
+DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+VERBOSE=false
+QUIET=false
+export VERBOSE
+export QUIET
+
+HTML_TEMP=${HTML_TEMP:-$DIR/.temp}
+export HTML_TEMP
+
+for arg in "$@"
+do
+  case $arg in
+    -h|--help)
+      echo "Usage: $0 [-h|--help]"
+      echo "Usage: $0 [-n|--no-update] [-q|--quiet] [-v|--verbose]"
+      echo
+      echo "  -h|--help       Show this usage statement."
+      echo "  -q|--quiet      Don't emit any messages except errors/warnings."
+      echo "  -v|--verbose    Show verbose output from every build step."
+      exit 0
+      ;;
+    -q|--quiet)
+      QUIET=true
+      VERBOSE=false
+      ;;
+    -v|--verbose)
+      VERBOSE=true
+      QUIET=false
+      set -vx
+      ;;
+    *)
+      ;;
+  esac
+done
+
+# XXX This should probably be factored out into a function
+if [ -z "$HTML_SOURCE" ]; then
+  $QUIET || echo "HTML_SOURCE environment variable not set..."
+  $QUIET || echo "OK, looking for HTML source file..."
+  PARENT_DIR=$(dirname $DIR)
+  if [ -f $PARENT_DIR/html/source ]; then
+    $QUIET || echo "OK, looked in the $PARENT_DIR/html directory and found HTML source there..."
+    HTML_SOURCE=$PARENT_DIR/html
+  else
+    if [ -f $DIR/html/source ]; then
+      $QUIET || echo "OK, looked in the html subdirectory here and found HTML source..."
+      HTML_SOURCE=$DIR/html
+    else
+      # TODO Before giving up, should we maybe also check $HOME/html? Or anywhere else?
+      $QUIET || echo "Didn't find the HTML source on your system..."
+      chooseRepo
+    fi
+  fi
+else
+  $QUIET || echo "HTML_SOURCE environment variable is set to $HTML_SOURCE; looking for HTML source..."
+  if [ -f "$HTML_SOURCE/source" ]; then
+    $QUIET || echo "OK, looked in the $HTML_SOURCE directory and found HTML source there..."
+  else
+    $QUIET || echo "Looked in the $HTML_SOURCE directory but didn't find HTML source there..."
+    unset HTML_SOURCE
+    chooseRepo
+  fi
+fi
+export HTML_SOURCE
+
+rm -rf $HTML_TEMP && mkdir -p $HTML_TEMP
+
+# Fetch unicode.xml
+$QUIET || echo "Downloading unicode.xml (can take a short time, depending on your bandwidth)...";
+curl --location $($VERBOSE && echo "-v") $($QUIET && echo "-s") \
+  https://www.w3.org/2003/entities/2007xml/unicode.xml.zip \
```

Comment:
https://raw.githubusercontent.com/w3c/xml-entities/gh-pages/unicode.xml is faster and sheds the unzip dependency.